### PR TITLE
Add duckdb support

### PR DESCRIPTION
**Pull request by clausherther**
## Summary of Changes

This PR adds support for duckdb, including updates to integration tests and CI.

## Why Do We Need These Changes
dbt-date recently added support for duckdb in conjunction with dbt metrics layer testing.  

## Reviewers
@jwills

### CODE REVIEW

File: integration_tests/models/schema_tests/timeseries_hourly.sql [hunk 1]
```diff
@@ -1,5 +1,9 @@
-{{ dbt_date.date_spine('hour',
-        start_date=dbt_date.n_days_ago(10),
-        end_date=dbt_date.tomorrow()
+{% set now = modules.datetime.datetime.now().replace(minute=0, hour=0, second=0, microsecond=0) %}
```
Comment (jwills) [inline integration_tests/models/schema_tests/timeseries_hourly.sql:1]: Maybe it needs to be:

`{% set now = modules.datetime.date.today() %}`

?

Comment (jwills) [reply integration_tests/models/schema_tests/timeseries_hourly.sql:1]: (Looking at the BQ error in CI)

**Review by jwills — APPROVED** [review]
LGTM-- thank you so much @clausherther!

File: integration_tests/models/schema_tests/timeseries_hourly.sql (hunk 1 above)
Comment (clausherther) [reply integration_tests/models/schema_tests/timeseries_hourly.sql:1]: Maybe, thanks! This is also related to https://github.com/calogica/dbt-date/pull/106 which I just merged. Once I figure out the rest of the BQ CI issues, I'll merge and release this early this week.

**Pull request merged**